backend/mem/fileSystem.go

```diff
@@ -0,0 +1,176 @@
+package mem
+
+import (
+	"path"
+	"time"
+
+	"github.com/c2fo/vfs/v5"
+	"github.com/c2fo/vfs/v5/backend"
+	"github.com/c2fo/vfs/v5/utils"
+)
+
+//Scheme defines the filesystem type.
+const Scheme = "mem"
+const name = "In-Memory Filesystem"
+
+// FileSystem implements vfs.Filesystem for the mem filesystem.
```

Comment:
This comment should be directly above the FileSystem struct, not with the obj and objMap between, also, be sure to follow the camel casing of FileSystem (vfs.FileSystem, not vfs.Filesystem) and because of that, all places where file system is spelled out, it should be two words. (see the line above, and line 12 and 14 also needs to be fixed.)